src/main/java/org/spongepowered/common/SpongeCatalogType.java

```diff
@@ -73,9 +66,34 @@ public final String toString() {
     }
 
     protected ToStringHelper toStringHelper() {
-        return Objects.toStringHelper(this)
-                .add("id", getId())
-                .add("name", getName());
+        return toStringHelper(this);
+    }
+
+    public static ToStringHelper toStringHelper(CatalogType type) {
+        return Objects.toStringHelper(type)
+                .add("id", type.getId())
+                .add("name", type.getName());
+    }
+
+    public static <T extends CatalogType & org.spongepowered.api.text.translation.Translatable> ToStringHelper toStringHelperTranslatable(T type) {
+        return toStringHelper(type)
+                .add("translation", type.getTranslation());
+    }
+
+    public static abstract class Named extends SpongeCatalogType {
+
+        private final String name;
+
+        public Named(String id, String name) {
+            super(id);
+            this.name = checkNotNull(name, "name");
+        }
+
+        @Override
+        public final String getName() {
+            return this.name;
+        }
+
     }
 
     public static abstract class Translatable extends SpongeCatalogType implements org.spongepowered.api.text.translation.Translatable {
```

Comment:
Rename this inner class, I can't stand the sight of clashing class names in this case.